### ISSUE TITLE
Windows 11

### ISSUE THREAD

**Issue by anl-do** [damienvanrobaeys/Run-in-Sandbox#16]
😢
When will RiS run on Windows 11?
I can't live without it anymore.

best regards
Andreas

**Comment by damienvanrobaeys** [comment]
Hi,

I haven't tested w11 yet.
Have you already tested runinsandbox on it ?

**Comment by anl-do** [comment]
I have upgraded my PC from 10 to 11. RiS was installed perviously and working fine. After the udate, the context menues are not visible anymore. However, the sandbox is still enabled in Win11.
When I run "Add_Structure.ps1" in Win11, the powershell is stuck.
![image](https://user-images.githubusercontent.com/82723836/136205956-914acb40-aa99-4095-ae1d-80544153d49e.png)

The log file says:
[10.06.21 14:46:42] - INFO : Starting the configuration of RunInSandbox
[10.06.21 14:46:42] - INFO : The script has been launched with admin rights
[10.06.21 14:46:43] - INFO : The sources folder exists

[10.06.21 14:46:43] - SUCCESS : Sources have been copied in C:\ProgramData\Run_in_Sandbox
[10.06.21 14:46:43] - SUCCESS : Sources files have been unblocked
[10.06.21 14:46:43] - INFO : Mapping registry HKCR
[10.06.21 14:46:43] - SUCCESS : Mapping registry HKCR
[10.06.21 14:46:43] - INFO : Exporting registry path: exefile

Even after minutes, there is no further progress.
I'm not a powershell expert. How can I debug? I can't see what exactly the script is trying to do when there is no progress.
Sorry, I'm no real help :-(

**Comment by damienvanrobaeys** [comment]
No worries :-)
Could you please check below registry key ? Is there Run PS1 in Sandbox ?
HKEY_CLASSES_ROOT\Microsoft.PowerShellScript.1\Shell

**Comment by damienvanrobaeys** [comment]
Which version did you test meaning when have you downloaded ris ?

**Comment by damienvanrobaeys** [comment]
nope, i will install a W11 tonight and try it.
I will send you a version tomorrow

**Comment by anl-do** [comment]
That sounds awesome! I really love it and cannot work without ris 😊.
If you need anything from me, just let me know. I wished I was a powershell wizard like you...and David Segura and Nickolaj Andersen and and and...
THANKS A LOT

**Comment by anl-do** [comment]
Hi Damien, I have tried the DEV version.
This one installs without issues.
![image](https://user-images.githubusercontent.com/82723836/136359376-e69f8b5a-145b-4b79-9ab8-3634fef7d43a.png)

Not all context menus work like expected:
- the .PS1 context menu does not appear at all
![image](https://user-images.githubusercontent.com/82723836/136359752-9d69c62b-c4e8-4aab-aa2e-a08aadcfaf99.png)
- the .intunewin part redirects all files and folders, where the selected intunewin file is located, to the sandbox
![image](https://user-images.githubusercontent.com/82723836/136360071-fa2bdd43-3c4b-4cb1-b43b-9388269f2942.png)
![image](https://user-images.githubusercontent.com/82723836/136360235-6e2a1d7c-9690-44d2-a4f2-8b9fabe1dac8.png)
- the command for the intunewin file is not executed in the sandbox (I have no screenshot for that)
- the MSI works fine
- the "extract the ZIP in sandbox" produces an error
![image](https://user-images.githubusercontent.com/82723836/136361228-09054ce7-e175-4204-951c-2f55caf173ca.png)
- the Regkey part works as expected
- the VBS part also works as expected

Hopefully, my tests will help you at least a little. I'm here for more tests. Use me like your labourer 😂.
As far as I'm concerned, this is now all I need for most of my usecases. On the other hand, I'm very interested in the code itself and which sections you edit. Maybe I can improve my skills a little...

**Comment by damienvanrobaeys** [comment]
Could you plz send me a mail at damien.vanrobaeys@gmail.com ?

**Comment by oMadMartigaNo** [comment]
I really appreciate that you are working on this problem @damienvanrobaeys Thanks!

**Comment by damienvanrobaeys** [comment]
> Hi Damien, I have tried the DEV version. This one installs without issues. ![image](https://user-images.githubusercontent.com/82723836/136359376-e69f8b5a-145b-4b79-9ab8-3634fef7d43a.png)
> 
> Not all context menus work like expected:
> 
> * the .PS1 context menu does not appear at all
>   ![image](https://user-images.githubusercontent.com/82723836/136359752-9d69c62b-c4e8-4aab-aa2e-a08aadcfaf99.png)
> * the .intunewin part redirects all files and folders, where the selected intunewin file is located, to the sandbox
>   ![image](https://user-images.githubusercontent.com/82723836/136360071-fa2bdd43-3c4b-4cb1-b43b-9388269f2942.png)
>   ![image](https://user-images.githubusercontent.com/82723836/136360235-6e2a1d7c-9690-44d2-a4f2-8b9fabe1dac8.png)
> * the command for the intunewin file is not executed in the sandbox (I have no screenshot for that)
> * the MSI works fine
> * the "extract the ZIP in sandbox" produces an error
>   ![image](https://user-images.githubusercontent.com/82723836/136361228-09054ce7-e175-4204-951c-2f55caf173ca.png)
> * the Regkey part works as expected
> * the VBS part also works as expected
> 
> Hopefully, my tests will help you at least a little. I'm here for more tests. Use me like your labourer 😂. As far as I'm concerned, this is now all I need for most of my usecases. On the other hand, I'm very interested in the code itself and which sections you edit. Maybe I can improve my skills a little...

Regarding the zip and intunexi part what was the full path of those files ?

**Comment by damienvanrobaeys** [comment]
I found something interesting. I am now able to create a conext menu for PS1.
I will update the runinsandox on Tuesday and will add also some new features:
- Run PS1 as user
- Run PS1 as system
- Run MSIX file
- Extract ISO file
- Run PPKG file
- Open URL or HTML file

**Comment by HarmVeenstra** [comment]
Yes! Works, contex-menu is there for everything.. Except.. For PS1 files, it gave me the error in the screenshot

![image](https://user-images.githubusercontent.com/5159742/141952187-bf9e800b-78ee-4034-a51e-4aa33c0c85a0.png)

**Comment by damienvanrobaeys** [comment]
What do you have in this key ? 
![image](https://user-images.githubusercontent.com/8832278/141955220-b6e07ade-16b4-4fbf-a592-e9d1a25f16c5.png)

**Comment by HarmVeenstra** [comment]
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Classes\.ps1]

[HKEY_CURRENT_USER\Software\Classes\.ps1\OpenWithProgids]
"VSCode.ps1"=""
"AppXxf01pj590w7z9mxmyv3nx0a9ewj3e51g"=hex(0):

**Comment by damienvanrobaeys** [comment]
no rOpenWithProgids ?

**Comment by HarmVeenstra** [comment]
Correct, this is the complete export from [HKEY_CURRENT_USER\Software\Classes.ps1]

**Comment by damienvanrobaeys** [comment]
ok, i will update the add_structure.ps1 in few minutes and will inform you when you can test

**Comment by HarmVeenstra** [comment]
Thank you!

**Comment by damienvanrobaeys** [comment]
Could you please download again sources and try again ?

**Comment by HarmVeenstra** [comment]
Still the same, value $Get_rOpenWithProgids_Default_Value is not set because (Get-Item "$Default_PS1_HKCU\rOpenWithProgids").Property fails. I don't have the rOpwnWithProgids value and it's not created by the script itself. Perhaps a detect and create if not exist?

**Comment by damienvanrobaeys** [comment]
and if you check in the registry for OpenWithProgids ?

**Comment by HarmVeenstra** [comment]
![image](https://user-images.githubusercontent.com/5159742/141972507-8d141e64-edfb-4077-9017-9641c7521f9d.png)

**Comment by damienvanrobaeys** [comment]
and if you go in the classes\vscode.ps1\shell ?

**Comment by HarmVeenstra** [comment]
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Classes\VSCode.ps1]
@="PowerShell Source File"
"AppUserModelID"="Microsoft.VisualStudioCode"

[HKEY_CURRENT_USER\Software\Classes\VSCode.ps1\DefaultIcon]
@="C:\\Users\\HarmV\\AppData\\Local\\Programs\\Microsoft VS Code\\resources\\app\\resources\\win32\\powershell.ico"

[HKEY_CURRENT_USER\Software\Classes\VSCode.ps1\shell]

[HKEY_CURRENT_USER\Software\Classes\VSCode.ps1\shell\open]
"Icon"="\"C:\\Users\\HarmV\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\""

[HKEY_CURRENT_USER\Software\Classes\VSCode.ps1\shell\open\command]
@="\"C:\\Users\\HarmV\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\" \"%1\""

**Comment by damienvanrobaeys** [comment]
ok i think i found why, you have another dword that i don't have

**Comment by HarmVeenstra** [comment]
Ok, my laptop is pretty standard I guess using Powershell 5 and 7 in conjunction with VScode ;)

**Comment by damienvanrobaeys** [comment]
could you please do a remove_structure, download again and try again ?

**Comment by HarmVeenstra** [comment]
![image](https://user-images.githubusercontent.com/5159742/141976541-92c7a659-3ad1-404b-a92d-a0fb039bcc72.png)

It works!

**Comment by damienvanrobaeys** [comment]
yihaaa, thanks for the feedback, that was really interesting

**Comment by HarmVeenstra** [comment]
No problem, glad to help and your tool is very usefull!

**Comment by damienvanrobaeys** [comment]
You're right
I am checking if device is w10 or 11 and then check location but it seems for some other reason some context menu may not work on some w11.
Do you have some issues ?

**Comment by anl-do** [comment]
I have not yet tested all.
If you agree, I'd close this "issue" here and open a new one as soon as I see something.
Regarding the very first question in this issue, we can consider it solved.
Thank you for your great efforts!!!! I can not repeat often enough.

**Comment by damienvanrobaeys** [comment]
With pleasure.
You're welcome.
And of course don't hesitate to open a new issue if needed 😁
